Tasks/Common/azure-arm-rest/azure-arm-mysql.ts

```diff
@@ -0,0 +1,230 @@
+            this.baseUri = baseUri;
+        }
+
+        if (options.apiVersion) {
+            this.apiVersion = options.apiVersion;
+        }
+        if (options.acceptLanguage) {
+            this.acceptLanguage = options.acceptLanguage;
+        }
+        if (options.longRunningOperationRetryTimeout) {
+            this.longRunningOperationRetryTimeout = options.longRunningOperationRetryTimeout;
+        }
+        if (options.generateClientRequestId) {
+            this.generateClientRequestId = options.generateClientRequestId;
+        }
+
+        this.firewallRules = new FirewallRules(this);
+        this.mysqlServers = new MysqlServers(this);
+    }
+}
+
+export class FirewallRules {
+    private client: AzureMysqlManagementClient;
+
+    constructor(client) {
+        this.client = client;
+    }
+    
+    /**
+     * Create or update firewall rule for mysql server 
+     * @param resourceGroupName     resource group name of mysql server 
+     * @param serverName            mysql server name
+     * @param firewallRuleName      rule name to be added or updated
+     * @param parameters            optional parameter like start and end ip address
+     * @param callback              response callback 
+     */
+    public createOrUpdate(resourceGroupName: string, serverName: string, firewallRuleName: string, parameters, callback?:  azureServiceClient.ApiCallback) {
+        var client = this.client;
+        if (!callback) {
+            throw new Error(tl.loc("CallbackCannotBeNull"));
+        }
+
+        // Validate
+        try {
+            this.client.isValidResourceGroupName(resourceGroupName);
+            if(!this.isNameValid(serverName)){
+                throw new Error(tl.loc("MysqlServerNameCannotBeEmpty"));
+            }
+            if(!this.isNameValid(firewallRuleName)){
+                throw new Error(tl.loc("FirewallRuleNameCannotBeNull"));
+            }
+        }
+        catch (error) {
+            return callback(error);
+        }
+
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'PUT';
+        httpRequest.headers = {};
+        httpRequest.uri = this.client.getRequestUri(
+            '//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/firewallRules/{firewallRuleName}',
+            {
+                '{resourceGroupName}': resourceGroupName,
+                '{serverName}': serverName,
+                '{firewallRuleName}': firewallRuleName
+            });
+            
+        if (parameters !== null && parameters !== undefined) {
+            httpRequest.body = JSON.stringify(parameters);
+        }
+
+        this.client.beginRequest(httpRequest).then(async (response) => {
+            var deferred = Q.defer<azureServiceClient.ApiResult>();
+            var statusCode = response.statusCode;
+            if (statusCode != 200 && statusCode != 201 && statusCode != 202) {
```

Comment:
@vincentdass - Can you review this to ensure we are covered on reliability based on learnings in Webapp task.

Reply:
In there documentation they don't have 204 response code.

Reply:
Now we are polling for 202.